### ISSUE TITLE
Batch execution problem in Python3?

### ISSUE THREAD

**Issue by beginnermind** [google/google-api-python-client#95]
```
def getThreads(email, userId):

    threadList = []

    def threadCallback(request_id, response, exception):
        if exception is not None:
            raise APIError("Error occurred during thread callback")
        threadList.append(response)

    batch = BatchHttpRequest()

    try:
        credential = EmailAddress.objects.get(email=email, user=userId).credential
        gmail_client = getGmailClient(credential)
        threads = gmail_client.users().threads().list(userId='me', maxResults=10, q="", labelIds=['INBOX']).execute()
    except ObjectDoesNotExist:
        logger.error(Template("Error retrieving credential for email: $email, user: $user").safe_substitute(email=email, user=userId))
        raise ObjectDoesNotExist
    except errors.HttpError as e:
        # Returned for invalid requests, e.g. maxResults=0
        logger.error(Template("Gmail API: An error occurred: $e").safe_substitute(e=e))
        raise APIError("Gmail API: HttpError")

    if 'threads' in threads:
        for thread in threads['threads']:
            # TODO: Cache threads on 'id' field
            batch.add(gmail_client.users().threads().get(userId='me', id=thread['id'], format='metadata', metadataHeaders=['subject','date','to','from']), callback=threadCallback)
        batch.execute()

    return threadList
```

This code runs fine in Python2.7, but in Python3.3 the batch.execute() line throws an error:

Error: Can't convert 'bytes' object to str implicitly
Location: python3.3/site-packages/googleapiclient/http.py in _execute, line 1260

This is using google-api-python-client==1.4.0.

Any ideas what's going on? Again the first non-batch call is working fine, it's only the batch call that's throwing an error.

**Comment by methane** [comment]
It may be fixed via 0915761 but it is not released yet.

**Comment by beginnermind** [comment]
Thanks, that fixed it!

**Comment by pferate** [comment]
@nathanielmanistaatgoogle, I think this issue can be closed, it was resolved in #77.

**Comment by nathanielmanistaatgoogle** [comment]
Thanks for pointing that out; closing.
